src/main/java/org/apache/sysds/runtime/controlprogram/paramserv/LocalPSWorker.java

```diff
@@ -89,19 +97,19 @@ private void computeEpoch(long dataSize, int batchIter) {
 			try {
 				for (int j = 0; j < batchIter; j++) {
 					ListObject gradients = computeGradients(params, dataSize, batchIter, i, j);
-	
+
 					boolean localUpdate = j < batchIter - 1;
-					
-					// Accumulate the intermediate gradients (async for overlap w/ model updates 
+
+					// Accumulate the intermediate gradients (async for overlap w/ model updates
 					// and gradient computation, sequential over gradient matrices to avoid deadlocks)
 					ListObject accGradientsPrev = accGradients.get();
 					accGradients = _tpool.submit(() -> ParamservUtils.accrueGradients(
```

Comment:
we only need to accrue gradients if we aim to exchange them - if model averaging is enabled this can be avoided.